target_postgres/connector.py

```diff
@@ -255,12 +262,22 @@ def _handle_array_type(self, jsonschema: dict) -> ARRAY | JSONB:
         # Case 3: tuples
         return ARRAY(JSONB()) if isinstance(items, list) else JSONB()
 
+    def _handle_integer_type(self, jsonschema: dict) -> SMALLINT | INTEGER | BIGINT:
+        """Handle integer type."""
+        if maximum := jsonschema.get("maximum"):
+            if maximum < 2**15:
+                return SMALLINT()
+            if maximum < 2**31:
+                return INTEGER()
+
+        return BIGINT()
+
     @cached_property
     def jsonschema_to_sql(self) -> JSONSchemaToSQL:
         """Return a JSONSchemaToSQL instance with custom type handling."""
         to_sql = JSONSchemaToPostgres(content_encoding=self.interpret_content_encoding)
         to_sql.fallback_type = TEXT
-        to_sql.register_type_handler("integer", BIGINT)
+        to_sql.register_type_handler("integer", self._handle_integer_type)
```

Comment:
pretty sure the answer is yes here, but would this also be able to parse the schema that is generated by `postgres-tap`:

```
{"type":["integer","null"]}
```

?